core/domain/story_domain.py

```diff
@@ -687,6 +687,71 @@ def update_story_contents_from_model(
         versioned_story_contents['story_contents'] = conversion_fn(
             versioned_story_contents['story_contents'])
 
+    @property
+    def node_ids(self):
+        """Returns a list of all the node IDs that are part of this
+        story.
+
+        Returns:
+            list(str). List of node IDs.
+        """
+        return [node.id for node in self.story_contents.nodes]
+
+    @property
+    def first_node_id(self):
+        """Returns the first element in the node list of the story, which
+           corresponds to the first node that the user would encounter, or if
+           the story is empty, returns None.
+
+        Returns:
+            str|None. The node ID of the first node,
+                or None if the story is empty.
+        """
+        if len(self.story_contents.nodes) > 0:
+            return self.story_contents.nodes[0].id
+        else:
+            return None
+
+    def get_next_node_id(self, completed_node_ids):
+        """Returns the first node id in the story that has not yet
+           been completed by the learner, or if the story is completed,
+           returns None.
+
+        Args:
+            completed_node_ids: list(str). List of completed node
+                ids.
+
+        Returns:
+            str|None. The node ID of the next node,
+                or None if the story is completed.
+        """
+        for node_id in self.node_ids:
+            if node_id not in completed_node_ids:
+                return node_id
+        return None
+
+    def get_next_node_id_in_sequence(self, current_node_id):
+        """Returns the node ID of the node just after the node
+           corresponding to the current node id. If the user is on the
+           last node, None is returned.
+
+        Args:
+            current_node_id: str. The id of node currently
+                completed.
+
+        Returns:
+            str|None. The node ID of the next node,
+                or None if the passed id is the last one in the story.
+        """
+        node_just_unlocked = None
+
+        for index in range(0, len(self.story_contents.nodes) - 1):
+            if self.story_contents.nodes[index].id == current_node_id:
```

Comment:
Ditto here, check destination_nodes of current node and in that, you can return first element in the list.

In the destination_node_id  list of a story node, you can assume the learner visits node at index 0 first and so on, but in the story as a whole, this is not the case, as mentioned in the previous comment.

So, I think it would be better to create a function that does this traversal and use that throughout.